File: бинарный.java
import java.util.PriorityQueue;

// Бинарная куча через PriorityQueue
PriorityQueue<Integer> minHeap = new PriorityQueue<>();
minHeap.add(5);
minHeap.add(2);
minHeap.add(10);
System.out.println(minHeap.poll()); // 2

// Биномиальная куча реализуется через список деревьев

File: бинарный.cpp
#include <queue>
#include <vector>

// Бинарная куча через priority_queue
std::priority_queue<int, std::vector<int>, std::greater<int>> minHeap;
minHeap.push(5);
minHeap.push(2);
minHeap.push(10);
int minVal = minHeap.top(); // 2
minHeap.pop();

// Реализация биномиальной кучи требует собственного класса с деревьями
